src/java.base/share/classes/sun/nio/ch/SocketChannelImpl.java

```diff
@@ -955,6 +955,7 @@ public boolean connect(SocketAddress remote) throws IOException {
             try {
                 writeLock.lock();
                 try {
+                    ensureOpen();
```

Comment:
Similar comment as that for `ServerSocketChannelImpl`, the `beginConnect()` call a few lines below already does a `ensureOpen()` after acquiring the `stateLock` monitor.